test/utils/TestKeyManager.sol

```diff
@@ -0,0 +1,120 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.23;
+
+import {KeyType, Key, KeyLib} from "../../src/libraries/KeyLib.sol";
+import {Test} from "forge-std/Test.sol";
+import {Vm} from "forge-std/Vm.sol";
+
+struct TestKey {
+    uint40 expiry;
+    KeyType keyType;
+    bool isSuperAdmin;
+    bytes publicKey;
+    // saved to sign messages
+    uint256 privateKey;
+}
```

Comment:
I wonder if it would be nicer to have 

struct TestKey {
Key key
uint256 privateKey
}

so we're working directly with the underlying Key struct

Reply:
I also expose like a `toKey()` function if we want the underlying Key object

it just might be awkward to do assertions about the underlying key then bc it would look like
`assertEq(testKey.key.expiry, 0)`

Reply:
ah yeah that makes sense, lets just make sure to update this struct if we change Key